front/src/features/books/booksSlice.tsx

```diff
@@ -0,0 +1,81 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import api from "api/index";
+import { initialState } from "./helperData";
+
+import { IfetchBooksData, IAddItemToBasket } from "../interfaces/index";
+
+export const booksSlice = createSlice({
+  name: "books",
+  initialState: initialState,
+  reducers: {
+    fetchBooksDataStarted: (state) => {
+      state.status = "inProgress";
+    },
+    fetchBooksDataSucceeded: (
+      state,
+      action: PayloadAction<IfetchBooksData>
+    ) => {
+      state.data = [...action.payload.data];
```

Comment:
state.data = action.payload

Nie musisz tworzyć specjalnie typu IfetchBooksData (btw. powinno być IFetchBooksData), jeżeli przekazujesz tylko jeden argument. Wtedy możesz uprościć i zrobić po prostu action: PayloadAction<Object[]>

I nie rób Object[], tylko jakieś FetchedBook[] - nie musisz deklarować w TS wszystkich pól, które ma obiekt z BE, tylko te, które potrzebujesz.

Reply:
Typ jest obiektem posiadającym 2 właściwości  dlatego zrobiłem typ;
 x = {
  data: [{id: 457,…]
  metadata: {page: 1,…}
} tak wygląda w orginale;
nie wiem czy dobrze zrozumiałem o co Ci chodzi bo Object[] chyba by tu nie zadziałał z takim responsem.
During